# Case Study on Testing of Hypothesis

### Input:-A company started to invest in digital marketing as a new way of their product promotions.For that they collected data and decided to carry out a study on it.

In [1]:
#importing the relevant libraries
import pandas as pd
import numpy as np
from scipy import stats

In [2]:
#reading the data
data=pd.read_csv(r'C:/Users/DELNA.RAPHEL/Desktop/DSA/Assignment_week_08/Sales_add.csv')
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 5 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Month                           22 non-null     object
 1   Region                          22 non-null     object
 2   Manager                         22 non-null     object
 3   Sales_before_digital_add(in $)  22 non-null     int64 
 4   Sales_After_digital_add(in $)   22 non-null     int64 
dtypes: int64(2), object(3)
memory usage: 1008.0+ bytes


In [3]:
data.head()

Month      Region      Manager  Sales_before_digital_add(in $)  \
0  Month-1  Region - A  Manager - A                          132921   
1  Month-2  Region - A  Manager - C                          149559   
2  Month-3  Region - B  Manager - A                          146278   
3  Month-4  Region - B  Manager - B                          152167   
4  Month-5  Region - C  Manager - B                          159525   

   Sales_After_digital_add(in $)  
0                         270390  
1                         223334  
2                         244243  
3                         231808  
4                         258402

#### ● The company wishes to clarify whether there is any increase in sales after stepping into digital marketing.
- consider sample1=Sales_before_digital_ad,
 sample2=Sales_After_digital_add

In [4]:
#checking the statistical values of the samples using describe function
data[['Sales_before_digital_add(in $)','Sales_After_digital_add(in $)']].describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

- sample size=22.
- mean of second sample is greater than the first one.i.e, there is a increase in sales before and after digital marketing.

###### *Hypothesis testing

*Null Hypothesis
H0 :- There is no increase in sales after digital marketing

*Alternate Hypothesis
H1:- There is an increase in sales after digital marketing

- Consider,
- Level of significance=5%

We are going with paired sample t-test because:

- The sample size is less than 30 and we are comparing the mean of the same group at different times(before and after advertisement.)


In [5]:
t_score,pval = stats.ttest_rel(data['Sales_before_digital_add(in $)'],data['Sales_After_digital_add(in $)'])
print('test statistic is',t_score)
print('p-value is',(pval))

#considering the level of significance=5%
if pval<0.05:
    print("\nReject null hypothesis")
else:
    print("\nAccept null hypothesis")


test statistic is -12.09070525287017
p-value is 6.336667004575778e-11

Reject null hypothesis


In [6]:
#degree of freedom
#df=n-1
df=data['Sales_before_digital_add(in $)'].count()-1
print('degree of freedom is',df)

#T critical value for a one-tailed test 
tcritical=stats.t.ppf(1-0.05,21)
print('tcritical is',tcritical)

degree of freedom is 21
tcritical is 1.7207429028118775


*The test statistic is -12.09070525287017.(-ve sign Indicates the mean of sample_2 is greater than mean of sample_1.The absolute value of test statistic is > than tcritical.

*p-value is 6.336667004575778e-11 is < than 0.05(significance level.) 

*Hence,we can reject null hypothesis and conclude, there is an increase in Sales before and after digital marketing.

#### ● The company needs to check whether there is any dependency between thefeatures “Region” and “Manager”.

We are going with Chi-Squared test for independence because:

-we need to check the dependency for categorical variables.

* Null Hypothesis:H0:There is no dependency between Region and Manager

* Alternative Hypothesis:There is dependency between Region and manager


In [7]:
#converting the data into a table with frequencies
observed_values= data.groupby('Region')['Manager'].value_counts().unstack()
observed_values

Manager     Manager - A  Manager - B  Manager - C
Region                                           
Region - A            4            3            3
Region - B            4            1            2
Region - C            1            3            1

This is also known as contingency table.It is a tabular representation of frequencies for particular combinations of values of two discrete random variables.

In [8]:
#Performing Chi-square test and
# finding Chi-Squared statistic,p-value,degree of freedom,Expected frequency.
c,p,dof,expected= stats.chi2_contingency(observed_values)

print('Chi-Squared statistic is',c)
print('p-value is',p)
print('degree of freedom is',df)#Degree of freedom=(row count-1)(column count-1)

#Expected frequency = (row sum * column sum) / table sum
print('\nexpected_values=\n',expected)

Chi-Squared statistic is 3.050566893424036
p-value is 0.5493991051158094
degree of freedom is 21

expected_values=
 [[4.09090909 3.18181818 2.72727273]
 [2.86363636 2.22727273 1.90909091]
 [2.04545455 1.59090909 1.36363636]]


In [9]:
chi2_critical_value = stats.chi2.ppf(1-0.05,4)# 5% significance level and df=4
print('chi2_critical_value is ',chi2_critical_value)

if(c>chi2_critical_value and p<0.05):
    print("\nReject null hypothesis")
else:
    print("\nAccept null hypothesis")

chi2_critical_value is  9.487729036781154

Accept null hypothesis


* There is no dependency between Region and Manager.